src/utils.jl

```diff
@@ -193,7 +193,7 @@ function iter_segments(args...)
 end
 
 # helpers to figure out if there are NaN values in a list of array types
-anynan(i::Int, args::Tuple) = any(a -> !isfinite(_cycle(a,i)), args)
+anynan(i::Int, args::Tuple) = any(a -> try !isfinite(_cycle(a,i)) catch MethodError false end, args)
```

Comment:
This seems really dirty to me? The issue that poped up was the when you tried to for loop over a `SegmentsIterator` with string values. It would check whether values are finite but `isfinite` isn't defined for strings. From what I understand this is only meant to check for `NaN` so it seems "fine" to me to give `false` when the values cannot be checked by `isfinite` because then it cannot be a `NaN`. But I'm open to suggestions on how to make this clearer.